frontend/src/lib/OutputArtifactLoader.test.ts

```diff
@@ -243,6 +244,44 @@ describe('OutputArtifactLoader', () => {
     });
   });
 
+  describe('buildMarkdownViewerConfig', () => {
+    it('requires "source" metadata field', () => {
+      const metadata = { header: 'header', format: 'format' };
+      expect(OutputArtifactLoader.buildMarkdownViewerConfig(metadata as any)).rejects.toThrowError(
+        'Malformed metadata, property "source" is required.');
+    });
+
+    it('returns a markdown viewer config with basic metadata for inline markdown', async () => {
+      const metadata = { source: '# some markdown here', storage: 'inline' };
+      expect(await OutputArtifactLoader.buildMarkdownViewerConfig(metadata as any)).toEqual({
+        markdownContent: '# some markdown here',
+        type: PlotType.MARKDOWN,
+      } as MarkdownViewerConfig);
+    });
+
+    it('returns a markdown viewer config with basic metadata for gcs path markdown', async () => {
+      const metadata = { source: 'gs://path', storage: 'gcs' };
+      fileToRead = `<html><body>
```

Comment:
Shouldn't this be defined with a const?

Reply:
This is a global variable defined in the test file so that the API mock will return it, and tests can change it freely. Probably not the cleanest solution but it's the most concise for tests.